src/config.rs

```diff
@@ -744,45 +744,32 @@ impl<'a> Cfg<'a> {
         &self,
         verbose: bool,
     ) -> Result<(LocalToolchainName, ActiveReason)> {
-        match self.find_override_config()? {
-            Some((override_config, reason)) => match override_config {
-                OverrideCfg::PathBased(path_based_name) => {
-                    let toolchain = Toolchain::with_reason(self, path_based_name.into(), &reason)?;
```

Comment:
Since we're only getting `LocalToolchainName` now, it's no longer necessary to construct `Toolchain` with `with_reason()`. Instead, `override_config.into_local_toolchain_name()` should be enough.

Reply:
Oops, actually not when a custom toolchain is active. I'll revert this for now...